ofproto/ofproto.c

```diff
@@ -5552,6 +5552,69 @@ handle_load_filter_prog(struct ofconn *ofconn, const struct ofp_header *oh)
     return error;
 }
 
+static enum ofperr
+handle_update_map(struct ofconn *ofconn, const struct ofp_header *oh)
+    OVS_EXCLUDED(ofproto_mutex)
+{
+    struct ofproto *ofproto = ofconn_get_ofproto(ofconn);
+    enum ofperr error;
+    error = reject_slave_controller(ofconn);
+    if (error) {
+        return error;
+    }
+
+    struct ol_update_map msg;
+    void *key = NULL;
+    void *value = NULL;
+    error = ofputil_decode_update_map(&msg, &key, &value, oh);
+    if (error) {
+        return error;
+    }
+
+    if(!msg.filter_prog) {
+        VLOG_WARN_RL(&rl,
+                     "The filter program identifier is not provided.");
+        return OFPERR_OFPBRC_EPERM;
+    }
+
+    struct ubpf_vm *vm = ubpf_vms_lookup(ofproto, msg.filter_prog);
+    if (!vm) {
+        VLOG_WARN_RL(&rl,
+                     "The referenced filter program could not be found.");
+        return OFPERR_OFPBRC_EPERM;
```

Comment:
In the future, we'll probably need our own set of error messages, but I can do that in a separate pull request.

Reply:
Agree, I've also recognized such a need.